mlrun/api/api/utils.py

```diff
@@ -512,6 +513,18 @@ def ensure_function_security_context(function, auth_info: mlrun.api.schemas.Auth
         SecurityContextEnrichmentModes.retain.value,
     ]:
 
+        # before iguazio 3.6 the user unix id is not passed in the session verification response headers
+        # so we need to request it explicitly
+        if (
+            auth_info.user_unix_id is None
+            and mlrun.api.utils.clients.iguazio.SessionPlanes.control
+            in auth_info.planes
+        ):
+            iguazio_client = mlrun.api.utils.clients.iguazio.Client()
+            auth_info.user_unix_id = iguazio_client.get_user_unix_id(auth_info.session)
+        else:
+            raise mlrun.errors.MLRunUnauthorizedError("Missing control plane session")
```

Comment:
You'll enter this `else` and raise an exception if you have the `user_unix_id` from the session verification.
Change the ifs around a bit so we don't raise an exception in this case.

Reply:
good catch 🙏

Reply:
FYI CI actually caught this error in the backport PR (not here because of the transient failure).